tests/bootstrap.php

```diff
@@ -32,28 +32,66 @@
 	$_test_root = '/tmp/wordpress-tests-lib';
 }
 
-// Force plugin to be active.
-$GLOBALS['wp_tests_options'] = array(
-	'active_plugins' => array( basename( TESTS_PLUGIN_DIR ) . '/load.php' ),
-);
-
-// Add filter to ensure the plugin's admin integration and all modules are loaded for tests.
 require_once $_test_root . '/includes/functions.php';
-tests_add_filter(
-	'plugins_loaded',
-	static function () {
-		require_once TESTS_PLUGIN_DIR . '/admin/load.php';
-		require_once TESTS_PLUGIN_DIR . '/admin/server-timing.php';
-		require_once TESTS_PLUGIN_DIR . '/admin/plugins.php';
-		$module_files = glob( TESTS_PLUGIN_DIR . '/modules/*/*/load.php' );
-		if ( $module_files ) {
-			foreach ( $module_files as $module_file ) {
-				require_once $module_file;
-			}
+
+// Check if we use the plugin's test suite. If so, disable the PL plugin and only load the requested plugin.
+$testsuite_count = array_count_values( $_SERVER['argv'] )['--testsuite'];
```

Comment:
What if someone invokes `phpunit` without providing a `--testsuite` arg? Won't this cause an error due to the array index not being defined. I don't think this check is needed since it is looping over the args below anyway. Also, shouldn't passing `--testsuite performance-lab` behave the same as if no `--testsuite` was provided at all? This doesn't seem to be happening right now, as the `else` condition won't be run. As I understand, the structure should rather be as follows:

```php
// Check if we use the plugin's test suite. If so, disable the PL plugin and only load the requested plugin.
$plugin_name = '';
foreach ( $_SERVER['argv'] as $index => $arg ) {
	if (
		'--testsuite' === $arg &&
		isset( $_SERVER['argv'][ $index + 1 ] ) &&
		'performance-lab' !== $_SERVER['argv'][ $index + 1 ] &&
		file_exists( TESTS_PLUGIN_DIR . '/plugins/' . $_SERVER['argv'][ $index + 1 ] )
	) {
		$plugin_name = $_SERVER['argv'][ $index + 1 ];
	}
}

if ( $plugin_name ) {
	$plugin_test_path = TESTS_PLUGIN_DIR . '/plugins/' . $plugin_name;
	tests_add_filter(
		'plugins_loaded',
		static function () use ( $plugin_test_path, $plugin_name ) {
			// Check if plugin has a "plugin/plugin.php" file.
			if ( file_exists( $plugin_test_path . '/' . $plugin_name . '.php' ) ) {
				require_once $plugin_test_path . '/' . $plugin_name . '.php';
			} elseif ( file_exists( $plugin_test_path . '/load.php' ) ) {
				// Check if plugin has a "plugin/load.php" file.
				require_once $plugin_test_path . '/load.php';
			} else {
				echo "Unable to locate standalone plugin bootstrap file in $plugin_test_path.";
				exit( 1 );
			}
		},
		1
	);
} else {
	// Force plugin to be active.
	$GLOBALS['wp_tests_options'] = array(
		'active_plugins' => array( basename( TESTS_PLUGIN_DIR ) . '/load.php' ),
	);

	// Add filter to ensure the plugin's admin integration and all modules are loaded for tests.
	tests_add_filter(
		'plugins_loaded',
		static function () {
			require_once TESTS_PLUGIN_DIR . '/admin/load.php';
			require_once TESTS_PLUGIN_DIR . '/admin/server-timing.php';
			require_once TESTS_PLUGIN_DIR . '/admin/plugins.php';
			$module_files = glob( TESTS_PLUGIN_DIR . '/modules/*/*/load.php' );
			if ( $module_files ) {
				foreach ( $module_files as $module_file ) {
					require_once $module_file;
				}
			}
		},
		1
	);
}
```

Reply:
Thanks. Updated in https://github.com/WordPress/performance/pull/972/commits/7f3b13b321ea1608412d8c4a83c7c73286ebc2fe